app/models/watchlist_instance.rb

```diff
@@ -342,14 +346,25 @@ def delete_watchlist_instance
   end
   # rubocop:enable Style/GuardClause
 
+  def self.remove_dropped_students(course, current_instances)
```

Comment:
nit: maybe some documentation to explain that it puts instances of dropped students into archived, returns not dropped students?